### PR TITLE
Add Group Normalization (GN) as an better alternative method for Batch Normalization (BN) in the situation of training with small batch size.

### PR DESCRIPTION
**Pull request by Min-Sheng**
## Status
**READY**

## Description
I added Group Normalization (GN) as an better alternative method for Batch Normalization (BN) in the situation of training with small batch size. If someone what to use it in convolution layer or deconvolution layer, just set "with_gn" parameter "True".

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Todos
- [ ] Tests
- [ ] Documentation

## Impacted Areas in Application
List general components of the application that this PR will affect:
* convolution.py
* deconvolution.py

### CODE REVIEW

**Comment by tomvars** [comment]
@fbragman Might be interesting for you

**Comment by Min-Sheng** [comment]
I found some bugs about option parameter and typo in my code and just fixed them.
